include/exadg/compressible_navier_stokes/spatial_discretization/operator.cpp

```diff
@@ -339,6 +349,124 @@ Operator<dim, Number>::prescribe_initial_conditions(VectorType & src, double con
   src = src_double;
 }
 
+template<int dim, typename Number>
+void
+Operator<dim, Number>::serialize_vectors(std::vector<VectorType const *> const & vectors) const
+{
+  std::vector<dealii::DoFHandler<dim> const *> dof_handlers{&dof_handler};
+  std::vector<std::vector<VectorType const *>> vectors_per_dof_handler{vectors};
+  if(param.restart_data.consider_mapping)
+  {
+    store_vectors_in_triangulation_and_serialize(param.restart_data.filename,
+                                                 vectors_per_dof_handler,
+                                                 dof_handlers,
+                                                 this->get_mapping(),
+                                                 &(*dof_handler_mapping),
+                                                 param.mapping_degree);
+  }
+  else
+  {
+    store_vectors_in_triangulation_and_serialize(param.restart_data.filename,
+                                                 vectors_per_dof_handler,
+                                                 dof_handlers);
+  }
+}
+
+template<int dim, typename Number>
+void
+Operator<dim, Number>::deserialize_vectors(std::vector<VectorType *> const & vectors)
+{
+  // Store ghost state to recover after deserialization.
+  std::vector<bool> const has_ghost_elements = get_ghost_state(vectors);
+
+  // Load potentially unfitting checkpoint triangulation of TriangulationType.
+  std::shared_ptr<dealii::Triangulation<dim>> checkpoint_triangulation =
+    deserialize_triangulation<dim>(dof_handler.get_triangulation(),
+                                   param.restart_data.filename,
+                                   param.restart_data.triangulation_type,
+                                   mpi_comm);
+
+  // Setup DoFHandlers *as checkpointed*, sequence matches `this->serialize_vectors()`.
+  dealii::DoFHandler<dim> checkpoint_dof_handler(*checkpoint_triangulation);
+  dealii::DoFHandler<dim> checkpoint_dof_handler_mapping(*checkpoint_triangulation);
```

Comment:
this and related has to be moved behind `if(consider_mapping)`

Reply:
we should also use `do_grid_to_grid_projection()` to reuse the `MatrixFree` object.
update this once #729 is merged

Reply:
both done :)